### PR TITLE
Add m0110a usb

### PR DESCRIPTION
**Pull request by elmo-space**
Added initial support for the drop-in replacement PCB for the Apple M0110a that I recently designed. The project is open source and is available under https://github.com/elmo-space/m0110a_usb

I have put the files in a new subdirectory named after my github user name (like it's done for many other projects). If that should be changed then please suggest a new subdirectory.
Please note that this is not to be confused with one of the m0110a converters. It is a completely new PCB design.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/elmospace/m0110a_usb/config.h [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0xA68C
+#define PRODUCT_ID 0xDB4B
+#define DEVICE_VER 0x0001
+#define MANUFACTURER APPLE
```
Comment (fauxpark) [inline keyboards/elmospace/m0110a_usb/config.h:26]: Not sure about this one...

File: keyboards/elmospace/m0110a_usb/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        KC_LCTL,  KC_LGUI,                  KC_SPC,                                      KC_BSLS, KC_LEFT, KC_RGHT, KC_DOWN,             KC_P0,           KC_PDOT, KC_PENT  \
+    ),
+    [1] = LAYOUT(
+        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_TRNS,    KC_INS,  KC_TRNS, KC_TRNS, KC_TRNS, \
+        KC_TRNS, KC_HOME, KC_UP,   KC_END,  KC_PGUP, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_HOME, KC_UP,   KC_PGUP, KC_TRNS, \
+        KC_TRNS, KC_LEFT, KC_DOWN, KC_RGHT, KC_PGDN, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_LEFT, KC_TRNS, KC_RGHT, KC_TRNS, \
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_END,  KC_DOWN, KC_PGDN,          \
+        KC_TRNS, KC_TRNS,                   KC_TRNS,                                     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_TRNS,          KC_TRNS, KC_TRNS  \
```
Comment (fauxpark) [inline keyboards/elmospace/m0110a_usb/keymaps/default/keymap.c:32]:
```suggestion
        KC_ESC,   KC_1,   KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,    KC_DEL, KC_EQL,  KC_PSLS, KC_PAST,
        KC_TAB,   KC_Q,   KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,             KC_P7,  KC_P8,   KC_P9,   KC_PMNS,
        KC_LCAP,  KC_A,   KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,              KC_P4,  KC_P5,   KC_P6,   KC_PPLS,
        KC_LSFT,  KC_Z,   KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, MO(1),   KC_UP,               KC_P1,  KC_P2,   KC_P3,
        KC_LCTL,  KC_LGUI,                  KC_SPC,                                      KC_BSLS, KC_LEFT, KC_RGHT, KC_DOWN,             KC_P0,           KC_PDOT, KC_PENT
    ),
    [1] = LAYOUT(
        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_TRNS,    KC_INS,  KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_HOME, KC_UP,   KC_END,  KC_PGUP, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_HOME, KC_UP,   KC_PGUP, KC_TRNS,
        KC_TRNS, KC_LEFT, KC_DOWN, KC_RGHT, KC_PGDN, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_LEFT, KC_TRNS, KC_RGHT, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_END,  KC_DOWN, KC_PGDN,
        KC_TRNS, KC_TRNS,                   KC_TRNS,                                     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_TRNS,          KC_TRNS, KC_TRNS
```

File: keyboards/elmospace/m0110a_usb/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        KC_TRNS, KC_TRNS,                   KC_TRNS,                                     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_TRNS,          KC_TRNS, KC_TRNS  \
+    ),
+    [3] = LAYOUT(
+        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_TRNS,    KC_INS,  KC_TRNS, KC_TRNS, KC_TRNS, \
+        KC_TRNS, KC_HOME, KC_UP,   KC_END,  KC_PGUP, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_HOME, KC_UP,   KC_PGUP, KC_TRNS, \
+        KC_TRNS, KC_LEFT, KC_DOWN, KC_RGHT, KC_PGDN, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_LEFT, KC_TRNS, KC_RGHT, KC_TRNS, \
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_END,  KC_DOWN, KC_PGDN,          \
+        KC_TRNS, KC_TRNS,                   KC_TRNS,                                     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_TRNS,          KC_TRNS, KC_TRNS  \
```
Comment (fauxpark) [inline keyboards/elmospace/m0110a_usb/keymaps/via/keymap.c:46]:
```suggestion
        KC_ESC,   KC_1,   KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC,    KC_DEL, KC_EQL,  KC_PSLS, KC_PAST,
        KC_TAB,   KC_Q,   KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,             KC_P7,  KC_P8,   KC_P9,   KC_PMNS,
        KC_LCAP,  KC_A,   KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_ENT,              KC_P4,  KC_P5,   KC_P6,   KC_PPLS,
        KC_LSFT,  KC_Z,   KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, MO(1),   KC_UP,               KC_P1,  KC_P2,   KC_P3,
        KC_LCTL,  KC_LGUI,                  KC_SPC,                                      KC_BSLS, KC_LEFT, KC_RGHT, KC_DOWN,             KC_P0,           KC_PDOT, KC_PENT
    ),
    [1] = LAYOUT(
        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_TRNS,    KC_INS,  KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_HOME, KC_UP,   KC_END,  KC_PGUP, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_HOME, KC_UP,   KC_PGUP, KC_TRNS,
        KC_TRNS, KC_LEFT, KC_DOWN, KC_RGHT, KC_PGDN, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_LEFT, KC_TRNS, KC_RGHT, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_END,  KC_DOWN, KC_PGDN,
        KC_TRNS, KC_TRNS,                   KC_TRNS,                                     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_TRNS,          KC_TRNS, KC_TRNS
    ),
    [2] = LAYOUT(
        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_TRNS,    KC_INS,  KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_HOME, KC_UP,   KC_END,  KC_PGUP, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_HOME, KC_UP,   KC_PGUP, KC_TRNS,
        KC_TRNS, KC_LEFT, KC_DOWN, KC_RGHT, KC_PGDN, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_LEFT, KC_TRNS, KC_RGHT, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_END,  KC_DOWN, KC_PGDN,
        KC_TRNS, KC_TRNS,                   KC_TRNS,                                     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_TRNS,          KC_TRNS, KC_TRNS 
    ),
    [3] = LAYOUT(
        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_TRNS,    KC_INS,  KC_TRNS, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_HOME, KC_UP,   KC_END,  KC_PGUP, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_HOME, KC_UP,   KC_PGUP, KC_TRNS,
        KC_TRNS, KC_LEFT, KC_DOWN, KC_RGHT, KC_PGDN, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_LEFT, KC_TRNS, KC_RGHT, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_END,  KC_DOWN, KC_PGDN,
        KC_TRNS, KC_TRNS,                   KC_TRNS,                                     KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,             KC_TRNS,          KC_TRNS, KC_TRNS
```

File: keyboards/elmospace/m0110a_usb/rules.mk [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (fauxpark) [inline keyboards/elmospace/m0110a_usb/rules.mk:35]:
```suggestion
```

File: keyboards/elmospace/m0110a_usb/readme.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+Drop in replacement PCB for the Apple M0110a
+
+* Keyboard Maintainer: [elmo-space](https://github.com/elmo-space)
+* Hardware Supported: M0110a USB
+* Hardware Availability: [Project is open source](https://github.com/elmo-space/m0110a_usb)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make m0110a_usb:default
```
Comment (zvecr) [inline keyboards/elmospace/m0110a_usb/readme.md:13]:
```suggestion
    make elmospace/m0110a_usb:default
```

File: keyboards/elmospace/m0110a_usb/info.json [hunk 6]
```diff
@@ -0,0 +1,89 @@
+{
+    "keyboard_name": "m0110a_usb",
+    "url": "",
+    "maintainer": "elmo-space",
+    "width": 3,
+    "height": 2,
```
Comment (zvecr) [inline keyboards/elmospace/m0110a_usb/info.json:6]:
```suggestion
    "width": 19.25,
    "height": 5,
```

File: keyboards/elmospace/m0110a_usb/config.h (hunk 1 above)
Comment (elmo-space) [reply keyboards/elmospace/m0110a_usb/config.h:26]: I just checked and the M0110a serial to usb converter also has Apple for the manufacturer in the config.h file so idk

https://github.com/qmk/qmk_firmware/blob/master/keyboards/converter/m0110_usb/config.h

Comment (fauxpark) [reply keyboards/elmospace/m0110a_usb/config.h:26]: Then that should really be changed too, I guess. But not in this PR obviously.

File: keyboards/elmospace/m0110a_usb/readme.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
+# m0110a_usb
+
+![m0110a_usb](https://i.imgur.com/h8nN8On.jpg)
```
Comment (noroadsleft) [inline keyboards/elmospace/m0110a_usb/readme.md:3]: Swaps this image for a smaller version.

```suggestion
![m0110a_usb](https://i.imgur.com/h8nN8Onl.jpg)
```

File: keyboards/elmospace/m0110a_usb/info.json [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+            {"label":"R", "x":4.5, "y":1}, 
+            {"label":"T", "x":5.5, "y":1}, 
+            {"label":"Y", "x":6.5, "y":1}, 
+            {"label":"U", "x":7.5, "y":1}, 
+            {"label":"I", "x":8.5, "y":1}, 
+            {"label":"O", "x":9.5, "y":1}, 
+            {"label":"P", "x":10.5, "y":1}, 
+            {"label":"{", "x":11.5, "y":1}, 
+            {"label":"}", "x":12.5, "y":1}, 
+            {"label":"Return", "x":13.5, "y":1, "h":2},
```
Comment (noroadsleft) [inline keyboards/elmospace/m0110a_usb/info.json:40]: Need to move this...

```suggestion
```

File: keyboards/elmospace/m0110a_usb/info.json [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+            {"label":"D", "x":3.75, "y":2}, 
+            {"label":"F", "x":4.75, "y":2}, 
+            {"label":"G", "x":5.75, "y":2}, 
+            {"label":"H", "x":6.75, "y":2}, 
+            {"label":"J", "x":7.75, "y":2}, 
+            {"label":"K", "x":8.75, "y":2}, 
+            {"label":"L", "x":9.75, "y":2}, 
+            {"label":":", "x":10.75, "y":2}, 
+            {"label":"\"", "x":11.75, "y":2}, 
+            {"label":"4", "x":15.25, "y":2},
```
Comment (noroadsleft) [inline keyboards/elmospace/m0110a_usb/info.json:57]: ... down here. `info.json` objects need to be in the same order used by `keymap.c`.

```suggestion
            {"label":"Return", "x":13.5, "y":1, "h":2}, 
            {"label":"4", "x":15.25, "y":2}, 
```

**Comment by elmo-space** [comment]
closing this for now, will reopen later. have to do some changes to the board and firmware first.

**Pull request closed without merging**